### PR TITLE
Variables inheritance

### PR DESCRIPTION
**Pull request by JulienTant**
This PR allows parent variables to be inherited by child commands

Something to consider: If we decide to move the injection of `os.Environ()` AFTER we inject the command variables, then we could have the server local variable take precedence over the ones in the files. I don't know if you guys have a preference, this could be changed in a later PR if needed. Here's the code we would change for reference: https://github.com/mattermost/ops-tool/blob/6f18f65ec0bac6f34faa0883271ba41003b76519/server/command.go#L77-L84
Basically the first line of this block would move after the loop at the end of the block.

Bonus:

I also made some changes to make the linter (`make lint`) happy with our code (mostly changing a few if/else to switch

### CODE REVIEW

**Review by pfltdv — APPROVED** [review]
Thanks alot Julien. This is great idea to move those variables to provider.
